enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,322 @@
+   may or may not be reconciled by the operator managing the resource.
+2. Updating tags of individual resources is not supported and any tag present in 
+   `.status.platformStatus.azure.resourceTags` of `infrastructure.config.openshift.io/v1` resource 
+   will result in adding tags to all Openshift managed Azure resources. 
+
+### Risks and Mitigations
+
+### Drawbacks
+- User-defined tags cannot be updated on an Azure resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in scope.
+  User-defined tags can be updated on the following Azure resources.
+    1. Virtual machine resources created for master and worker nodes.
+    2. Storage account
+    3. NetworkEdge
+    4. Internal Registry
+
+- OpenShift is bound to have the common limitation for all Azure resources created
+  by it and constraints other resources with the least matching limit as below
+    1. Tag names cannot have `microsoft`, `azure`, `windows` prefixes which are
+       reserved for Azure use.
+    2. An Azure storage account has a limit of 128 characters for the tag name.
+    3. An Azure DNS zone or Traffic or Front Door resource tag name cannot have spaces,
+       special/unicode characters or start with a number.
+    4. An Azure Automation or Content Delivery Network or DNS resource can have a
+       maximum of 15 tags.
+
+- Administrator will have to manually perform below tags pertaining actions
+    1. removing the undesired tags from the required resources.
+    2. update tag values of the required resources.
+    3. update tags of the resources which are not managed by an operator.
+    4. update tags of the resources for which update logic is not supported by an operator.
+
+## Design Details
+
+### Open Questions
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+- Feature available for end-to-end usage.
+- Complete end user documentation.
+- UTs and e2e tests are present.
+- Gather feedback from the users.
+
+#### Tech Preview -> GA
+N/A. This feature is for Tech Preview, until decided for GA.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+On upgrade:
+- Cluster operators that update the tags of Azure resources created for cluster 
+  should refer the new fields and take action. Any new resource created post-upgrade
+  and the operators managing the resource will add the user defined tags to the 
+  resource. But the same does not apply to already existing resources, components may 
+  or may not update the resources with the user defined tags.
+
+On downgrade:
+- The status field may remain populated, components may or may not continue 
+  to tag newly created resources with the additional tags depending on regardless of
+  whether given component still has logic to respect the status tags, after the downgrade.
```

Comment:
Is it "depending on" or is it "regardless of"?

I don't understand this sentence.  Are you saying that components might already have some logic to add tags besides the ones in `userTags`, and those components may apply those tags after downgrade?

Reply:
It should be "depending on", I will correct it.
Added this statement to be on the safe side, should any operator have the logic to persist userTags in their own object and use it.